### ISSUE TITLE
How do you use SURF in the latest opencv version.

### ISSUE THREAD

**Issue by ratom** [WangHewei16/DMCNet-for-Video-Engagement-Understanding#1]
I got this error everytime when using SURF 
----> 1 surf = cv2.xfeatures2d.SURF_create(extended=1)

AttributeError: module 'cv2.cv2' has no attribute 'xfeatures2d'

Can you help me

**Comment by WangHewei16** [comment]
Hello, you likely installed cv2 using pip, it does not include any patented modules like SURF. This is the sole approach to accessing cv2.xfeatures. SURF running is to establish from src with the contrib modules and the OPENCV_ENABLE_NONFREE cmake flag set. Since OpenCV4.4.0/3.4.11, cv2.features2d.SIFT_create() is also available to utilize SURF. Wish you all the best in your research study :)

**Comment by WangHewei16** [comment]
Hello, you likely installed cv2 using pip, it does not include any patented
modules like SURF. This is the sole approach to accessing cv2.xfeatures.
SURF running is to establish from src with the contrib modules and the
OPENCV_ENABLE_NONFREE cmake flag set. Since OpenCV4.4.0/3.4.11,
cv2.features2d.SIFT_create() is also available to utilize SURF. Wish you
all the best in your research study :)

**Comment by WangHewei16** [comment]
> I got this error everytime when using SURF ----> 1 surf = cv2.xfeatures2d.SURF_create(extended=1)
> 
> AttributeError: module 'cv2.cv2' has no attribute 'xfeatures2d'
> 
> Can you help me

Hello, you likely installed cv2 using pip, it does not include any patented modules like SURF. This is the sole approach to accessing cv2.xfeatures. SURF running is to establish from src with the contrib modules and the OPENCV_ENABLE_NONFREE cmake flag set. Since OpenCV4.4.0/3.4.11, cv2.features2d.SIFT_create() is also available to utilize SURF. Wish you all the best in your research study :)

**Comment by ratom** [comment]
Thanks for your suggestion. But I am a beginner. Do you mean to use cv2.xfeatures. SURF.

**Comment by WangHewei16** [comment]
> 

Yeah! You can try it. Our version of OpenCV or other libraries is different. Our laboratory conducted this research one year before. Hence, you could try to use the method suitable for the version on your desktop. Especially these patented and non-free libraries, they have different limitations in different versions/access methods. I believe you will have a good start in this fascinating field.

**Comment by WangHewei16** [comment]
> Thanks for your suggestion. But I am a beginner. Do you mean to use cv2.xfeatures. SURF.

Yeah! You can try it. Our version of OpenCV or other libraries is different. Our laboratory conducted this research one year before. Hence, you could try to use the method suitable for the version on your desktop. Especially these patented and non-free libraries, they have different limitations in different versions/access methods. I believe you will have a good start in this fascinating field.

**Comment by ratom** [comment]
Thanks. Can you tell me which version of OpenCV you used. I will use the same version. I have no issue with SIFT , it is working perfectly but, got problem with SURF. 
Kind regards

**Comment by WangHewei16** [comment]
> Thanks. Can you tell me which version of OpenCV you used. I will use the same version. I have no issue with SIFT , it is working perfectly but, got problem with SURF. Kind regards

You could check any configuration version in the ReadMe file in this repository. Since then, our lab server has also undergone updates for this library, so I did not record the specific details version number each time. My suggestion is you could check the most suitable ameliorating method based on your version from some coding-solving website such as Stack Overflow. If you know how to find the right solving approach based on your configuration, your future research study will go pretty smoothly.
